edx_exams/apps/core/management/commands/bulk_add_course_staff.py

```diff
@@ -84,11 +84,11 @@ def add_course_staff_from_csv(self, csv_file, batch_size, batch_delay):
         # bulk create course staff
         for i in range(0, len(reader), batch_size):
             CourseStaffRole.objects.bulk_create(
-                CourseStaffRole(
+                (CourseStaffRole(
                     user=User.objects.get(username=row.get('username')),
                     course_id=row.get('course_id'),
                     role=row.get('role'),
-                )
-                for row in reader[i:i + batch_size]
+                ) for row in reader[i:i + batch_size]),
+                ignore_conflicts=True,
```

Comment:
Question: is it possible to just use `ignore_conflicts=True` here rather than taking the more convoluted approach above for users, see below for code snippet (where we first create a list of users that need to be created and don't exist, then use this for the bulk operation). 

Looking at the documentation, it seems like ignore_conflicts isn't completely supported for mysql: https://docs.djangoproject.com/en/5.0/ref/models/querysets/#bulk-create. But this line sugg

It is working when I test locally but this must be because we are using sqlite for local/test, which is supported. Oracle/MySQL does not support but I also see this. I am wondering if ignore_conflicts wasn't used before because of exactly this. It seems like this could have been the reason for taking the other approach in the first place but I just want to be sure.

> Warning

> On MySQL and MariaDB, setting the ignore_conflicts parameter to True turns certain types of errors, other than duplicate key, into warnings. Even with Strict Mode. For example: invalid values or non-nullable violations. See the [MySQL documentation](https://dev.mysql.com/doc/refman/en/sql-mode.html#ignore-strict-comparison) and [MariaDB documentation](https://mariadb.com/kb/en/ignore/) for more details.

```
users_to_create = []
        users_existing = {u.username for u in User.objects.filter(username__in=[r.get('username') for r in reader])}
        for row in reader:
            if row.get('username') not in users_existing:
                users_to_create.append(row)
                users_existing.add(row.get('username'))
        # bulk create users
        for i in range(0, len(users_to_create), batch_size):
            User.objects.bulk_create(
                User(
                    username=user.get('username'),
                    email=user.get('email'),
                )
                for user in users_to_create[i:i + batch_size]
            )
            time.sleep(batch_delay)
```

Reply:
After speaking about this in standup, it looks like we can use ignore_conflicts. Tested locally and found other usages of ignore_conflicts within bulk_create in other places in the platform. So should be good to go here.